### PR TITLE
docs: various fixes.

### PR DESCRIPTION
**Pull request by neilconway**
* Document `records_per_epoch`
* Add some additional release notes for 0.12.13
* Various other fixes

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/reference/experiment-config.txt [hunk 1]
```diff
@@ -66,17 +66,18 @@ Top-Level Fields
...
-  a decision based on a validation of a trial, to maintain consistency
-  in the event of a failure.
+  checkpointed:
+
+  1. During training, a model may be checkpointed to allow the trial's execution
+     to be suspended and later resumed on a different Determined agent
+  2. When the trial's experiment is completed, to allow the resulting model to
+     be exported from Determined (e.g., for deployment)
+  3. Before the search method makes a decision based on a validation of a trial,
+     to maintain consistency in the event of a failure.
```
Comment (stoksc) [inline docs/reference/experiment-config.txt:80]: more of a question than a change request or anything, should this have a period at the end or not?

File: docs/reference/experiment-config.txt [hunk 2]
```diff
@@ -668,7 +683,7 @@ more suitable for large experiments with many trials.
   hyperparameter configuration.
 
 ``max_length``
-  The maximum training length of any one trial, in terms of records, batches or epochs
+  The maximum training length of any one trial, in terms of records, batches, or epochs
```
Comment (stoksc) [inline docs/reference/experiment-config.txt:686]: I take it we always use serial commas? Do we have a grammar style guide, ha?

File: docs/reference/experiment-config.txt [hunk 3]
```diff
@@ -1071,19 +1088,28 @@ details on how to set environment variables in containers.
...
-training units: records, batches or epochs. For example, to specify the ``max_length`` for a
-searcher in terms of batches, the configuration would read as shown below.
+Some configuration settings, such as searcher training lengths and budgets,
+``min_validation_period``, and ``min_checkpoint_period``, can be expressed in
+terms of a few training units: records, batches, or epochs. A *record* is a
+single labeled example. A *batch* is a group of records (the number of records
+in a batch is configured via the ``global_batch_size`` hyperparameter). An
+*epoch* is a single copy of the entire training data set; the number of records
+in an epoch is configured via the :ref:`records_per_epoch
+<config-records-per-epoch>` configuration field.
```
Comment (stoksc) [inline docs/reference/experiment-config.txt:1098]: praise: this is much more descriptive than the original.

File: docs/reference/experiment-config.txt (hunk 1 above)
Comment (neilconway) [reply docs/reference/experiment-config.txt:80]: Good catch, I think periods would be good -- or at any rate being consistent.

File: docs/reference/experiment-config.txt (hunk 2 above)
Comment (neilconway) [reply docs/reference/experiment-config.txt:686]: I don't think we have a grammar style guide but I personally prefer serial commas.

**Pull request merged**